app/browser/movie-search.mjs

```diff
@@ -1,5 +1,75 @@
 /* globals document HTMLElement customElements */
 import formatTitle from '../lib/formatTitle.mjs'
+import MoviePosterHTML from '../elements/movie-poster.mjs'
+import StarRatingHTML from '../elements/star-rating.mjs'
+const api = {
+  store: {}
+}
+
+const mix = (superclass) => new Mixer(superclass)
+
+class Mixer {
+  constructor(superclass) {
+    this.superclass = superclass
+  }
+
+  with(...mixins) {
+    return mixins.reduce((c, mixin) => mixin(c), this.superclass)
+  }
+}
+
+export default class EnhanceBase extends HTMLElement {
+  constructor() {
+    super()
+    this.api = api || {}
+    this.store = this.api?.store || {}
+    this.context = {}
+  }
+
+  get state() {
+    const attrs = this.attributes.length
+      ? this.attrsToObject(this.attributes)
+      : {}
+
+    return {
+      attrs,
+      context: this.context,
+      store: this.store
+    }
+  }
+
+  attrsToObject(attrs = []) {
+    const attrsObj = {}
+    for (let d = attrs.length - 1; d >= 0; d--) {
+      let attr = attrs[d]
+      attrsObj[attr.nodeName] = attr.nodeValue
+    }
+    return attrsObj
+  }
+
+  html(strings, ...values) {
+    const collect = []
+    for (let i = 0; i < strings.length - 1; i++) {
+      collect.push(strings[i], values[i])
+    }
+    collect.push(strings[strings.length - 1])
+    return collect.join('')
+  }
+}
+
+const CustomElementMixin = (superclass) => class extends superclass {
+  constructor() {
+    super()
+    this.template.content.querySelectorAll('style')
```

Comment:
Removes the style and script tags from the template only if it's a Custom Element.
This allows us to leave them in if we decide to use Shadow DOM and need the styles to remain in the template.